core/arch/arm/mm/core_mmu.c

```diff
@@ -1595,21 +1595,22 @@ static void *phys_to_virt_tee_ram(paddr_t pa)
 	uint32_t a;
 	paddr_t p;
 
-	if (pa >= CFG_TEE_LOAD_ADDR && pa < get_linear_map_end())
+	if (pa >= CFG_TEE_LOAD_ADDR &&
+	    pa < virt_to_phys((void *)get_linear_map_end()))
```

Comment:
I would think more about preserving current implementation:
```c
if (pa >= CFG_TEE_LOAD_ADDR && pa < get_linear_map_end())
	return (void *)(vaddr_t)pa;
```
Once linear mapping is no more mandated, these 2 lines can be removed.